egs/swbd/s5c/local/reverberate/reverberate_data_dir.py

```diff
@@ -0,0 +1,165 @@
+    parser.add_argument("--noise-list", type=str, default = None,
+                        help="Noise information file")
+    parser.add_argument("--num-replications", type=int, dest = "num_replica", default = 1,
+                        help="Number of replicate to generated for the data")
+    parser.add_argument("input_dir",
+                        help="Input data directory")
+    parser.add_argument("output_dir",
+                        help="Output data directory")
+
+    print(' '.join(sys.argv))
+
+    args = parser.parse_args()
+    args = CheckArgs(args)
+
+    return args
+
+def CheckArgs(args):
+    if not os.path.exists(args.output_dir):
+        os.makedirs(args.output_dir)
+
+    ## Check arguments.
+    if args.rir_list is None:
+        raise Exception("Rir information file must be provided")
+    
+    if not os.path.isfile(args.rir_list):
+        raise Exception(args.rir_list + "not found")
+    
+    if args.noise_list is not None:
+        if not os.path.isfile(args.noise_list):
+            raise Exception(args.noise_list + "not found")
+
+    return args
+
+
+def Corrupt_wav(input_dir, output_dir, rir_list, num_replica, prefix):
+    rirs = list_cyclic_iterator(return_nonempty_lines(open(rir_list).readlines()), random_seed = 1)
+    wav_files = open(input_dir + "/wav.scp", 'r').readlines()
+    command_list = []
+    for i in range(num_replica):
+        for j in range(len(wav_files)):
+            split1 = wav_files[j].split()
+            wav_pipe = " ".join(split1[1:])
+            output_wav_id = prefix + str(i) + "_" + split1[0]
+            rir = rirs.next().split()
+            rir_file_location = rir[-1]
+            command_list.append("{2} {0} wav-reverberate - {1} - |\n".format(wav_pipe, rir_file_location, output_wav_id))
+
+    file_handle = open(output_dir + "/wav.scp", 'w')
+    file_handle.write("".join(command_list))
+    file_handle.close()
+
+
+def Corrupt_wav_noise(input_dir, output_dir, rir_list, noise_list, snr_array, num_replica, prefix):
+    rirs = list_cyclic_iterator(return_nonempty_lines(open(rir_list).readlines()), random_seed = 1)
+    noises = list_cyclic_iterator(return_nonempty_lines(open(noise_list).readlines()), random_seed = 1)
+    snrs = list_cyclic_iterator(snr_array)
+    wav_files = open(input_dir + "/wav.scp", 'r').readlines()
+    command_list = []
+    for i in range(num_replica):
+        for j in range(len(wav_files)):
+            split1 = wav_files[j].split()
+            wav_pipe = " ".join(split1[1:])
+            output_wav_id = prefix + str(i) + "_" + split1[0]
+            rir = rirs.next().split()
+            rir_file_location = rir[-1]
+            noise = noises.next().split()
+            noise_file_location = noise[-1]
+            snr = snrs.next()
+            command_list.append("{4} {0} wav-reverberate --noise-file={2} --snr-db={3} - {1} - |\n".format(wav_pipe, rir_file_location, noise_file_location, snr, output_wav_id))
```

Comment:
On Wed, Apr 6, 2016 at 8:55 PM, Vimal Manohar <notifications@github.com>
wrote:

> In egs/swbd/s5c/local/reverberate/reverberate_data_dir.py
> <https://github.com/kaldi-asr/kaldi/pull/667#discussion_r58816432>:
>
> > +    rirs = list_cyclic_iterator(return_nonempty_lines(open(rir_list).readlines()), random_seed = 1)
> > +    noises = list_cyclic_iterator(return_nonempty_lines(open(noise_list).readlines()), random_seed = 1)
> > +    snrs = list_cyclic_iterator(snr_array)
> > +    wav_files = open(input_dir + "/wav.scp", 'r').readlines()
> > +    command_list = []
> > +    for i in range(num_replica):
> > +        for j in range(len(wav_files)):
> > +            split1 = wav_files[j].split()
> > +            wav_pipe = " ".join(split1[1:])
> > +            output_wav_id = prefix + str(i) + "_" + split1[0]
> > +            rir = rirs.next().split()
> > +            rir_file_location = rir[-1]
> > +            noise = noises.next().split()
> > +            noise_file_location = noise[-1]
> > +            snr = snrs.next()
> > +            command_list.append("{4} {0} wav-reverberate --noise-file={2} --snr-db={3} - {1} - |\n".format(wav_pipe, rir_file_location, noise_file_location, snr, output_wav_id))
>
> From Vijay's example:
> wav-reverberate --multi-channel-output=false \
> --rir-channel 1 \
> --point-source=noise1.wav --rir=rir1.wav --snr-db=10\
> --point-source=noise2.wav --rir=rir2.wav --snr-db=5 \
> --isotropic-noise=noise3.wav --snr-db=20 \
> --normalize-output=true \
> speech_rir.wav speech.wav
> The binary must support all these options. The lines for randomly picking
> noise, rir and snr must be modified to decide the number of different rir
> and noises (both foreground and background) and then decide the snr level
> for each noise.
>

Vimal, I think the discussion of the usage moved on quite bit since that
example.  Kaldi options parsing does not support multiple instances of the
same option, or position-dependent meaning of options.  We came up with an
alternative usage that differs in a bunch of respects.. I don't recall
right now where the thread is.

Dan


> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/667/files/1683e258c9f571271668ab6e1d2b0898e7b7ed33#r58816432>
>